File: file-management-system/src/domain/labels/Label.ts
/** Label — Flyweight 共享實體。建立後透過 LabelFactory.getOrCreate() 凍結，嚴格不可變。*/
export class Label {
  constructor(
    public readonly id: string,
    public readonly name: string,
    public readonly color: string,
    public readonly description: string,
    public readonly createdAt: Date,
  ) {}
}

File: backend/app/schemas/label.py
"""Pydantic schemas for Label request / response."""
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class LabelCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    color: str = Field(..., min_length=1, max_length=20)
    description: str = Field(default="", max_length=500)


class LabelResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    color: str
    description: str
    created_at: datetime
